packages/app/cypress/e2e/integration/runs.spec.ts

```diff
@@ -1,80 +1,229 @@
+            if (proj.runs?.nodes) {
+              proj.runs.nodes = []
+            }
+          }
+        }
+
+        return obj.result
+      })
+
+      cy.visitApp()
+      cy.get('[href="#/runs"]').click()
+      cy.findByText(defaultMessages.runs.connect.buttonProject).should('exist')
     })
 
-    cy.loginUser()
-    cy.visitApp()
+    it('displays how to record prompt when connected and no runs', () => {
+      cy.withCtx(async (ctx) => {
+        await ctx.actions.file.writeFileInProject('cypress.config.js', 'module.exports = {projectId: \'abcdef\'}')
+      })
+
+      cy.loginUser()
+      cy.remoteGraphQLIntercept(async (obj) => {
+        if (obj.result.data?.cloudProjectsBySlugs) {
+          for (const proj of obj.result.data.cloudProjectsBySlugs) {
+            if (proj.runs?.nodes) {
+              proj.runs.nodes = []
+            }
+          }
+        }
+
+        return obj.result
+      })
 
-    cy.get('[href="#/runs"]').click()
-    cy.contains('Connect your project').should('exist')
+      cy.visitApp()
+      cy.get('[href="#/runs"]').click()
+      cy.contains(defaultMessages.runs.empty.title)
+      cy.contains(defaultMessages.runs.empty.description)
+      cy.contains('--record --key 2aaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa')
+      cy.get('[data-cy="copy-button"]').last().click()
+      cy.contains('Copied!')
+    })
   })
 
-  it('if logged in and connected', { viewportWidth: 1200 }, () => {
-    cy.loginUser()
-    cy.visitApp()
-    cy.intercept('mutation-ExternalLink_OpenExternal', { 'data': { 'openExternal': true } }).as('OpenExternal')
+  context('Runs - Runs List', () => {
+    it('displays a list of recorded runs if a run has been recorded', () => {
+      cy.loginUser()
+      cy.visitApp()
+      cy.get('[href="#/runs"]').click()
+      cy.get('[data-cy="runs"]')
+    })
+
+    it('displays each run with correct information', () => {
+      cy.loginUser()
+      cy.visitApp()
+      cy.get('[href="#/runs"]').click()
+
+      cy.get('[href="http://dummy.cypress.io/runs/0"]').first().within(() => {
+        cy.findByText('fix: make gql work CANCELLED')
+        cy.get('[data-cy="run-card-icon"]')
+      })
+
+      cy.get('[href="http://dummy.cypress.io/runs/1"]').first().within(() => {
+        cy.findByText('fix: make gql work ERRORED')
+        cy.get('[data-cy="run-card-icon"]')
+      })
 
-    cy.get('[href="#/runs"]').click()
-    cy.contains('a', 'OVERLIMIT').click()
-    cy.wait('@OpenExternal')
-    .its('request.body.variables.url')
-    .should('equal', 'http://dummy.cypress.io/runs/4')
+      cy.get('[href="http://dummy.cypress.io/runs/2"]').first().within(() => {
+        cy.findByText('fix: make gql work FAILED')
+        cy.get('[data-cy="run-card-icon"]')
+      })
+
+      cy.get('[href="http://dummy.cypress.io/runs/0"]').first().as('firstRun')
+
+      cy.get('@firstRun').get('[data-cy="run-card-author"]').contains('John Appleseed')
```

Comment:
You could use `within` here as well to group all these queries under the firstRun card without using an alias, but it won't make a functional difference. I have to assume within does something very similar to this itself.